language-design.md

```diff
@@ -38,18 +44,17 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
-
+In the lab this week, we were given very limited time in which to work. Some of us tried to plan everything before we modified the code and therefore had very little time to implement our ideas. Other people began modifying the code, without first developing a plan for the growth and modification and so the changes they made were not always well thought out. The lab forced each of us to wrestle with how to divide our time between planning and implementing--a question that language developers must continually wrestle with. “Languages have now reached that large size where they can not be designed all at once, much less built all at once” [Steele, 1998]. We must plan the broad pattern and then alternate between planning and implementing so that we do not get so caught up in one or the other that we lose sight of the big picture. The broad pattern give us a consistency across methods without requiring the detail of a full plan for how we will implement everything. As Steele wrote, “A pattern should give hints or clues as to when and where it is best put to use” [Steele 1998]. The code we were given had no pattern, the methods seemed to take arbitrary arguments, each one stood separately and there was no fluid way to put them together. Each of us intuitively realized that it was not good. We had different ideas about what the pattern should be, but everyone began modifying the code to make function calls and parameters follow a more consistent, intuitive pattern. In doing so, we were mainly growing the language by changing the rules rather than by changing the vocabulary, the two ways to grow a language that Steele discusses, [Steele, 1998]. We were not trying to develop whole new methods, but rather trying to modify the input and output for many of the existing methods and only then perhaps add to the vocabulary.
```

Comment:
Now that I've read this, I think that the insights here answer the question Rebekah asked in her critique of Daniel's and my response to this prompt. After we worked separately, we then came together, but since we hadn't really discussed beforehand we found that we all had pretty different and maybe incompatible approaches. We were too deep in the bazaar approach, resulting in a lack of structure and cohesion, as you said happens in the response above this one. It also occurs to me, though, that the real issue may not have been so much that we were using the bazaar approach as a group, but that many of us (including myself) were like one-programmer bazaars. The approach of not planning ahead too much and just seeing what works by banging out some code seems to me like a sort of individual-level equivalent of the bazaar, where a bunch of people bang out code and someone picks out all the good stuff.